thorin/check.cpp

```diff
@@ -42,16 +59,35 @@ const Def* Infer::find(const Def* def) {
     return res;
 }
 
+bool Infer::eliminate(Array<Ref*> refs) {
+    if (std::ranges::any_of(refs, [](auto pref) { return should_eliminate(*pref); })) {
+        auto& world = (*refs.front())->world();
+        InferRewriter rw(world);
+        for (size_t i = 0, e = refs.size(); i != e; ++i) {
+            auto ref = *refs[i];
+            *refs[i] = ref->has_dep(Dep::Infer) ? rw.rewrite(ref) : ref;
+        }
+        return true;
+    }
+    return false;
+}
 /*
  * Check
  */
 
+#ifdef THORIN_ENABLE_CHECKS
```

Comment:
Isn't there missing a case to define `bool Check::fail` if `THORIN_ENABLE_CHECKS` isn't defined?

Reply:
This is inline in the header